src/gui/guiScrollContainer.cpp

```diff
@@ -67,6 +67,37 @@ void GUIScrollContainer::draw()
 	}
 }
 
+void GUIScrollContainer::setScrollBar(GUIScrollBar *scrollbar)
+{
+	m_scrollbar = scrollbar;
+
+	if (m_scrollbar && m_content_padding_px.has_value() && m_scrollfactor != 0.0f) {
+		// Set the scrollbar max value based on the content size.
+
+		// Get content size based on elements
+		core::rect<s32> size;
+		for (gui::IGUIElement *e : Children) {
+			core::rect<s32> abs_rect = e->getAbsolutePosition();
+			size.addInternalPoint(abs_rect.LowerRightCorner);
+		}
+
+		s32 hidden_content_px = *m_content_padding_px + (
+			m_orientation == VERTICAL
+				? (size.LowerRightCorner.Y - AbsoluteClippingRect.LowerRightCorner.Y)
+				: (size.LowerRightCorner.X - AbsoluteClippingRect.LowerRightCorner.X)
+		);
+		m_scrollbar->setMin(0);
+		m_scrollbar->setMax(std::ceil(hidden_content_px / std::fabs(m_scrollfactor)));
+		m_scrollbar->setPageSize(
+			m_orientation == VERTICAL
+				? (size.LowerRightCorner.Y - AbsoluteClippingRect.UpperLeftCorner.Y)
+				: (size.LowerRightCorner.X - AbsoluteClippingRect.UpperLeftCorner.X)
+		);
```

Comment:
Page size is (well, it's a bit weird, I don't understand the exact meaning (see `GUIScrollBar::setPosRaw`)), it is certainly not the inner scroll container size.
For reproduction, I've tried setting the padding really high, but the thumb didn't get small.

Reply:
`+ m_content_padding_px` was missing.

Reply:
I see now why this works. The page size calculation assumes the scrollbar has the same size as the outer container size.
So, this will break if scrollbar is another size than the scrollcontainer, or if it's differently oriented.

To fix this, you can multiply the page size value by `<scrollbar relative rect size (depending on scrollbar orientation)> / <scrollcontainer outer size (depending on scrollcontainer orientation)>`.